src/Language/Spectacle/Spec/Base.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE FunctionalDependencies #-}
+
+module Language.Spectacle.Spec.Base
+  ( Specifiable,
+    Fairness (Unfair, WeaklyFair, StronglyFair),
+    Modality (ModalAlways, ModalEventually, ModalUpUntil, ModalStaysAs, ModalInfinitelyOften),
+  )
+where
+
+import Data.Hashable (Hashable)
+
+import Data.Type.Rec (Rec, ReflectRow)
+
+-- ---------------------------------------------------------------------------------------------------------------------
+
+type Specifiable ctx =
+  ( Show (Rec ctx)
+  , Ord (Rec ctx)
+  , Hashable (Rec ctx)
+  , ReflectRow ctx
+  )
+
+data Fairness
```

Comment:
I know documentation is coming so don't take this comment as blocking feedback.

I think this data-type could use a comment that at minimum links to the TLA+ definition of fairness or explains it on the type.